### ISSUE TITLE
RAM usage difference in Linux and Windows

### ISSUE THREAD

**Issue by gullalc** [jwyang/faster-rcnn.pytorch#149]
Recently, I compiled faster-rcnn on windows and ran it on CPU. For some reason there is a huge spike in RAM when image is passed from the model while testing. On the other hand, it takes very less memory on Linux. The difference is more than 2 GB. Following is a part of the memory profile, which I did to know what was really going on. 

**Windows:**
  ```
   Line  Mem Usage    Increment
   215  276.762 MiB    0.000 MiB       if args.cuda > 0:
   216                                     checkpoint = torch.load(load_name)
   217                                 else:
   218  642.500 MiB  365.738 MiB           checkpoint = torch.load(load_name, map_location=(lambda storage, loc: storage))
   307  667.812 MiB    0.000 MiB               det_tic = time.time()
   308
   309                                         rois, cls_prob, bbox_pred, \
   310                                         rpn_loss_cls, rpn_loss_box, \
   311                                         RCNN_loss_cls, RCNN_loss_bbox, \
   312 3131.266 MiB 2463.453 MiB               rois_label = fasterRCNN(im_data, im_info, gt_boxes, num_boxes)
   313
   314 3131.266 MiB    0.000 MiB               scores = cls_prob.data
   315 3131.266 MiB    0.000 MiB               boxes = rois.data[:, :, 1:5]
```

**Linux**
```
   214  391.527 MiB    0.000 MiB       if args.cuda > 0:
   215                                     checkpoint = torch.load(load_name)
   216                                 else:
   217  756.840 MiB  365.312 MiB           checkpoint = torch.load(load_name, map_location=(lambda storage, loc: storage))
   306  854.660 MiB    0.000 MiB               det_tic = time.time()
   307                             
   308                                         rois, cls_prob, bbox_pred, \
   309                                         rpn_loss_cls, rpn_loss_box, \
   310                                         RCNN_loss_cls, RCNN_loss_bbox, \
   311  939.379 MiB   84.719 MiB               rois_label = fasterRCNN(im_data, im_info, gt_boxes, num_boxes)
   312                             
   313  939.379 MiB    0.000 MiB               scores = cls_prob.data
   314  939.379 MiB    0.000 MiB               boxes = rois.data[:, :, 1:5]
```

@jwyang - Did you guys try compiling it on Windows and faced such an issue?

**Comment by jwyang** [comment]
Hi, @gullalc , we did not compile it on windows, but I guess it is because you are running all on CPU on windows, while on GPU on the linux. That should be the reason.

**Comment by gullalc** [comment]
Both tests are for CPU only on both windows and Linux.

**Comment by gullalc** [comment]
@jwyang - The issue was because of the pytorch version. Windows is officially supported in pytorch v. 0.4. So, I was using the latest version. 

`volatile=True` is no more supported in the latest version, due to which all the gradients were being computed and stored during the call. 
This can be fixed by placing the call to the model inside the block `with torch.no_grad()`

**Comment by KirtoXX** [comment]
hi,i'm building on win10 too,but i filled on mask.sh,can u show  your ways?

**Comment by gullalc** [comment]
@KirtoXX - Follow the CFFI extension section on this [page](https://pytorch.org/docs/master/notes/windows.html#extension) to convert the C files to C++ for ROI Align, ROI crop, ROI pooling and NMS, and build each module separately. Once the CPU C++ files compile, you should have no problem running it on a CPU. 

I have tried only the CPU version on windows. Haven't been yet able to compile GPU cuda files on windows.
